### DISCUSSION TITLE
Implementing SONATA Ordering strategy

### DISCUSSION THREAD

**Discussion by cgfaportela** [MEF-GIT/MEF-LSO-Sonata-SDK#178] [Team Posts]
Dears,

We (as Proximus) are now planning to integrate SONATA Ordering on our portfolio of exposed API's to wholesale. As SONATA is being continuously improved, we would like to know what's the best strategy to implement it on the coming months. My first question would be if we could guide ourselves by the TMF622 latest version (R19.0.1) to implement the API envolope phase. My other question would be regarding the product specs (payload) that MEF is defining: there is a new draft (eline and UNI json) being worked on, is there any target date for it to be released? If there's no target yet for it, would it be recommended for us to guide ourselves by the existing 57.1 defined specs? 

Thanks in advance for your help!

**Comment by MichalLaczynskiAmartus** [comment]
Hi Carlos,

Welcome to the Community :)

At this moment there are differences between the TMF622 and Sonata Ordering. The API which is in SDK release 5 is a work-in-progress version. At this moment there works to deliver the next version of MEF 57.2 to update the business requirements for Ordering. Only after that, it will be possible to update the API specs. The best would be to wait for an IIS, yet there is no open project for it at the moment. Additionally, there is a project working on MEF-TMF API Alignment. You can check it here:
https://wiki.mef.net/pages/viewpage.action?pageId=106608028
Unfortunately, I am not able to list or name all differences and elaborate on them. Maybe @bigludo7 can help?

As for the product specifications - they are being worked within MEF W106 which is planned to be part of SDK release 5 on June 26th. I think you can start with existing and switch to new once they show up.

Michał

**Comment by bigludo7** [comment]
Hello Carlos,
Not a lot to add to Michał answer.
There is not an easy answer at this time.
MEF productOrder was based on TMF API. But, we had to a couple of MEF specific attribute in the API that were not part of the TMF offering. Also we restricted TMF to fit strictly MEF requirement (removed some attributes, defines search criteria, restrict notification type, etc...). These gaps are documented in a couple of documents that you'll find here: https://wiki.mef.net/display/LSO/Adhoc+-+LSO+API+Structure+and+Documentation+Contributions 

We are working jointly with TMF to define a set of rules to explain how to extend/restrict TMF API and fit MEF requirements. This work is in progress and I do not have - at me level - a date when this work will be over. It is probably a mid term objectif.

In short term we plan to update current MEF API to be aligned with 57.2.

**Comment by cgfaportela** [comment]
Hello Ludovic, Michał,

If I've understood it right, we should not rely ourselves on the TMF622 API Spec to build the Sonata Product Ordering API. At the same time, the MEF IIS definition for Sonata ProdOrder (that should overrule TMF622 API Spec) is still ongoing.. In other words, you suggest us to wait for the business requirements from 57.2 and a valid IIS so that we could start implementing the API's at our side. The IIS doesn't have yet an expected time to be ready at MEF side, I suppose. Therefore, it would consequently delay our previously expected timelines for delivering this API. I guess that the SDK release 5 also depends on these two MEF under work components and is also with no target for being finalized. 

About the IIS, I see that there are weekly calls going on. Would it be possible to join and follow (or even collaborate) for the work in progress?

In case that we would like to start our work before you release these two items, would you recommend us to take the SDK r4 or TMF622 Specifications (and json product UNI/eLine specs) into a first phase? Would it be a high impact on adapting the API later with your upcoming changes for IIS and 57.2?

Thanks for your help,
Carlos

**Comment by MichalLaczynskiAmartus** [comment]
Hi Carlos,

You can check the plan for Sonata releases on this page:
https://wiki.mef.net/display/CTO/LSO+Sonata+SDK+Releases

Unfortunately, MEF 57.2 will not be published even as a draft until release 5. Also, there are no pending works to deliver the IIS for this function. 
Of course, you are welcomed to join the existing calls for works on IIS for POQ (MEF W87) and Quote (MEF W115), from Proximus Patrick Roosen is a declared contributor there. 
https://wiki.mef.net/display/LSO/Sonata+IIS+-+2020-Q3+Call+Notes

If you are in big need of Order IIS maybe you could think of leading a project to deliver it?

I would suggest starting with MEF current work-in-progress version of the API. It will be updated to 57.2 and then I think that there will be fewer changes than to start with TMF and adopt to MEF. But that is my private point of view.

Best regards,
Michał

**Comment by cgfaportela** [comment]
Hi Michał,

Thanks for your feedback. 

Indeed, I know that Patrick Roosen is following but he is focused more on the API business requirements for Proximus Wholesale. I am looking into the IT architecture behind these requirements and how these API's should be integrated in our BSS/OSS solution. I will try to join on the coming sessions to understand/collaborate on the ongoing work for POQ and Quote then. 

I would be happy to collaborate on the Order IIS project but not lead it. At the moment, I am leading other internal IT projects in OSS and I dont think I have the expected and required time to head these discussions. Additionally, it is still not clear to me why we will work on a IIS that will differ from the TMF622 design. I saw the link from Ludovic where a few documents are listed regarding TMF and MEF challenges and divergences in the API's. However, I didn't find anything specific to SONATA that would answer to my questions (why running away from TMF622). I probably missed those ones..

Small remark on the Order IIS: if there's no project yet for that, I guess the announced target June 2020 for SDK 5 will be missed. Am I right? Looking to the release timeline, on SDK 5, a new IIS and Product Spec should be released together with the BR&UC 57.1. (https://wiki.mef.net/display/CTO/LSO+Sonata+SDK+Releases)

Now, summarizing your personal suggestion: we should implement SDK 4 that is mature at MEF and do not implement or guide ourselves by TMF622 spec (R19.0.1). Later on, when SDK 5 is out, it should be a light impact to support it.

Best regards,
Carlos

**Comment by MichalLaczynskiAmartus** [comment]
Hi Carlos,

The IIS for Ordering is not a deliverable for SDK 5 hence it will not affect it.

SDK5 is 1,5 month ahead and the POQ IIS is already in CfC thus you can enter the extended repo and check the Pull Request to know how will it change. Some steps pushing the APIs closer to TMF will be introduced already. For example, using MapCharacteristic to describe the MEF product specification. After CfC for POQ is passed, the rest of the APIs will be also changed to align with those changes.

Regards,
Michał

**Comment by cgfaportela** [comment]
Hi Michał,

I've understood it differently based on the LSO Sonata SDK release [page](https://wiki.mef.net/display/CTO/LSO+Sonata+SDK+Releases) where "underlined items indicate a release deliverable that is a delta from the previous SDK release". Then, may I suggest us to update the page to remove the underline on the ProdOrdering IIS for the SDK5? And I would guess the same for removing the underline on the product specs on SDK6 for ProdOrdering.
(got your clarification via the Sonata IIS Call.. thanks Michał)

Having your update, I see that SDK5 becomes feasible!

Thanks a lot,
Carlos

**Comment by MichalLaczynskiAmartus** [comment]
Carlos,

The IIS comes with 2 parts. The API definitions and the standard document ("Spec"). For SDK 5 there is only one MEF W87 Spec planned, for SDK6 the MEF W115. In the meantime, we will update the API to reflect changes introduced in MEF W87 and other BR&UC documents.

Michał
